### PR TITLE
drop_from_file and yaml_config for bedshift and unittests

### PR DESCRIPTION
**Pull request by berkuva**

### CODE REVIEW

File: bedshift/bedshift.py [hunk 1]
```diff
@@ -368,26 +376,75 @@ def drop(self, droprate):
...
 
+    def _find_intersection(self, fp, threshold=1):
+        """
+        find intersecting regions between the reference bedfile and the comparison file provided in the yaml config file
+
+        :param str fp: the filepath to the other bedfile containing regions to be compared to the reference bedfile
+        :param int threshold: the number of base pairs overlap needed for a match
+        :return dataframe intersect_ref: the dataframe consisting of matching regions, where regions are 
+        """
+        reference_bed = pybedtools.example_bedtool(os.path.abspath(self.bedfile_path))
```
Comment (aaron-gu) [inline bedshift/bedshift.py]: `NotImplementedError: "intersectBed" does not appear to be installed or on the path, so this method is disabled.  Please install a more recent version of BEDTools and re-import to use this method.`

This seems like a nonstarter. `pip install pybedtools` isn't enough to get it running

Comment (aaron-gu) [reply bedshift/bedshift.py]: What if we check for if BedTools is installed,

```
try:
    pybedtools.intersect(a, b)
except NotImplementedError:
    exact_matches(a, b)
```

File: bedshift/bedshift.py [hunk 2]
```diff
@@ -438,7 +502,152 @@ def read_bed(self, bedfile_path, delimiter='\t'):
...
+
+    :param float fp: the path to the configuration file for multiple add_from_file and drop_from_file
+    :return int: loaded yaml data
+    """
+    with open(fp, "r") as yaml_file:
+        config_data = yaml.load(yaml_file, Loader=yaml.FullLoader)
+    print("Loaded configuration settings from {}".format(fp))
+    return config_data
+
+def handle_yaml(bedshifter, yaml_fp):
```
Comment (aaron-gu) [inline bedshift/bedshift.py]: `handle_yaml` and `_read_from_yaml` should be `Bedshift` class methods.

Comment (aaron-gu) [reply bedshift/bedshift.py]: this function should also have a docstring (once it is moved into Bedshift class), so documentation will be automatically added to the documentation site.

File: bedshift/bedshift.py (hunk 1 above)
Comment (berkuva) [reply bedshift/bedshift.py]: Can you install pybedtools and try again?

Comment (aaron-gu) [reply bedshift/bedshift.py]: Did anything change?

**Comment by nsheff** [comment]
I changed this to merge into dev;  dev -> master PRs are reserved for releases.

**Comment by nsheff** [comment]
is this ready to merge to dev? I want to start testing the config file option

**Comment by aaron-gu** [comment]
@hyunjaecho94 I just resolved merge conflicts. Your final call to merge.

**Pull request merged**